examples/userapi/05_conditionaldimension.ipynb

```diff
@@ -0,0 +1,729 @@
+      "  struct timeval start_section0, end_section0;\n",
+      "  gettimeofday(&start_section0, NULL);\n",
+      "  /* Begin section0 */\n",
+      "  #pragma omp parallel num_threads(nthreads)\n",
+      "  {\n",
+      "    #pragma omp for collapse(1) schedule(static,1)\n",
+      "    for (int x = x_m; x <= x_M; x += 1)\n",
+      "    {\n",
+      "      #pragma omp simd aligned(f:32)\n",
+      "      for (int y = y_m; y <= y_M; y += 1)\n",
+      "      {\n",
+      "        f[x + 1][y + 1] = f[x + 1][y + 1] + 1;\n",
+      "      }\n",
+      "    }\n",
+      "  }\n",
+      "  /* End section0 */\n",
+      "  gettimeofday(&end_section0, NULL);\n",
+      "  timers->section0 += (double)(end_section0.tv_sec-start_section0.tv_sec)+(double)(end_section0.tv_usec-start_section0.tv_usec)/1000000;\n",
+      "  return 0;\n",
+      "}\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(op.ccode) # Print the generated code"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Devito API for relationals\n",
+    "\n",
+    "In order to construct a ConditionalDimension, one needs to get familiar with relationals. The Devito API for relationals currently supports the following classes of relations, overriding the [SymPy ones](https://docs.sympy.org/latest/modules/core.html#module-sympy.core.relational):\n",
+    "\n",
+    "- Le (less-than or equal)\n",
+    "- Lt (less-than)\n",
+    "- Ge (greater-than or equal)\n",
+    "- Gt (greater-than)\n",
+    "- Ne (not equal)\n",
+    "\n",
+    "Relationals are used to define a condition and are passed as an argument to Conditional Dimension at construction time.\n",
+    "\n",
+    "\n",
+    "# Devito API for ConditionalDimension\n",
+    "\n",
+    "When we define a ConditionalDimension we should provide the parent dimension and the condition to satisfy.\n",
+    "\n",
+    "`ci = ConditionalDimension(name, parent, condition)`\n",
+    "\n",
+    "i.e. \n",
+    "\n",
+    "`ci = ConditionalDimension(name='ci', parent=y, condition=Ge(f, 0))`\n",
+    "That means that our conditional dimension is nested under the `y` dimension and evaluates whether `f(x,y)>=0`.\n",
+    "\n",
+    "\n",
+    "## Defining a condition\n",
+    "Now, it's time to show a more descriptive example. Let's define a conditional that updates only the values of the grid that are larger than 0, f(x,y)>0.  We name our Conditional Dimension `cy` (conditional y).\n",
+    "When a ConditionalDimension is constructed we pass as an argument the parent dimension and the condition to satisfy. In this example we want to update again by one all the elements that are larger than 0, $f(x,y) > 0$\n",
+    "Before updating the elements we initialize our data again to the eye function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
```

Comment:
I don't think there's any value in displaying here the whole docstring -- just add a link to our documentation no ?

Reply:
or perhaps if you wanna keep it, maybe that's fine (@rhodrin ), but to be moved to the subsection above, not here which feels a bit too late

Reply:
I think @rhodrin mentioned that we can/could do that in the notebook.I also like the doc occurrence in tutorial 01_dsl the same way.